scripts/navbar_menu.js

```diff
@@ -0,0 +1,33 @@
+/* exported toggleMenu */
+
+// Toggle a menu contents to visible and rotate the menu icon
+function toggleMenu(menuId) {
+    let menu = document.getElementById(menuId);
+    [menu.getElementsByClassName("menu-contents")].forEach(element => {
+        element[0].classList.toggle("invisible");
+        element[0].classList.toggle("transparent");
+    });
+    [menu.getElementsByClassName("menu-icon")].forEach(element => {
+        element[0].classList.toggle("rotate");
+    });
+}
+
+// Close all dropdown menus if the user clicks outside of one
+document.addEventListener("click", function (event) {
+    let inMenu = false;
+    document.querySelectorAll(".menu").forEach(element => {
+        if (element.contains(event.target) || element == event.target) {
+            inMenu = true;
+        }
+    });
+
+    if (!inMenu) {
+        document.querySelectorAll(".menu-contents").forEach(element => {
+            element.classList.add('invisible');
+            element.classList.add('transparent');
```

Comment:
does this work for things like preventing tabbing to these elements?

Reply:
I don't know. In the browser how do you access something via tab that isn't an input? I'm using Firefox and clicking into the window and hitting tab isn't letting me navigate around.

Reply:
I was referring to child elements (anchors), but it seems you set them to `visibility: hidden`, so that should make them not tabbable.